CHANGELOG.md

```diff
@@ -14,6 +14,10 @@
   - New methods `Color::rgb_linear` and `Color::rgba_linear` will accept colors already in linear sRGB (the old behavior)
   - Individual color-components must now be accessed through setters and getters: `.r`, `.g`, `.b`, `.a`, `.set_r`, `.set_g`, `.set_b`, `.set_a`, and the corresponding methods with the `*_linear` suffix.
 - Despawning an entity multiple times causes a debug-level log message to be emitted instead of a panic [649] [651]
+- Breaking Change: Migrated to rodio 0.12, this means:
+  - All audio must be played on the main thread
```

Comment:
Would it be possible to "schedule" audio from other threads? That way you are not forced to use thread_local_systems to trigger sound effects etc?

Reply:
My idea was to use events and have an internal thread_local_system to handle those events, which would also (theoretically) allow pausing and modifying already playing sounds, but I didn't implement that during the initial migration. Should I include that as a commit? Also, do you think events *would* be a good idea? We could also probably have an `AudioScheduler` struct if you think that would work better than events.

Reply:
I'm not familiar enough with the Bevy architecture of events so can't really comment on events vs struct without seeing code. As for whether we need a solution in this PR, I would lean towards yes as to not break the ergonomics of being able to play audio from any system. It would be good to get some feedback from others though! I was just passing by to use your commit for testing iOS audio.

Reply:
Yeah I don't want to require thread local systems to queue up audio and I think we should resolve that before merging. How about something like this:

```rust
fn system(asset_server: Res<AssetServer>, audio: Res<Audio>) {
  let music = asset_server.load("song.mp3");
  audio.play(music);
}
```

Where `Audio` is the interface most users use to interact with Audio playback + audio configuration. AudioOutput would still be thread-local, but its job is to play queued sounds from Audio.